services/api/Tweek.ApiService.NetCore/Startup.cs

```diff
@@ -57,8 +61,10 @@ public void ConfigureServices(IServiceCollection services)
         {
 
             services.RegisterAddonServices(Configuration);
-
-            services.Decorate<IContextDriver>((driver, provider) => new TimedContextDriver(driver, provider.GetService<IMetrics>()));
+            services.Decorate<IContextDriver>((driver, provider) => {
```

Comment:
Agree. But, as this provider is optional - I thought it might be better that way - so we can just use the regular decorator if this provider is disabled.

Reply:
I thought of:
```
services.Decorate<IContextDriver>((driver, provider) =>  CreateChildDriver(provider, driver));
services.Decorate<IContextDriver>((driver, provider) => new TimedContextDriver(driver, provider.GetService<IMetrics>()))
```
CreateChildDriver can also be renamed to AddContextValidation.
That way you can easily disable each of the features. (validation, monitoring)
Not sure if it's possible to use decorate multiple times though.

Recomposing it with extension methods could also be nice, but it's just sugar:
```
services.Decorate<IContextDriver>((driver, provider) =>  driver.AddValidation(provider));
services.Decorate<IContextDriver>((driver, provider) => driver.AddMonitoring(provider.GetService<IMetrics>()))
```
or:
```
services.Decorate<IContextDriver>((driver, provider) =>  
driver.AddValidation(provider)
          .AddMonitoring(provider.GetService<IMetrics>()));
```

Reply:
Surprisingly, that worked. DotNet is pretty cool :)